packages/wrangler/src/__tests__/configuration.pages.test.ts

```diff
@@ -0,0 +1,785 @@
+							},
+						],
+						analytics_engine_datasets: [
+							{
+								binding: "TEST_AED_BINDING",
+							},
+						],
+						ai: {
+							binding: "TEST_AI_BINDING",
+						},
+					})
+				);
+			});
+
+			it("should return config corresponding to the named environment, if it exists in the configuration file", () => {
+				/*========================*/
+				/*        "preview"       */
+				/*========================*/
+				let { config, diagnostics } = normalizeAndValidateConfig(
+					pagesRawConfig,
+					undefined,
+					{
+						env: "preview",
+					}
+				);
+
+				expect(diagnostics.hasWarnings()).toBe(false);
+				expect(diagnostics.hasErrors()).toBe(false);
+				expect(config).toEqual(
+					expect.objectContaining({
+						/** TOP-LEVEL ONLY FIELDS **/
+						pages_build_output_dir: path.resolve(process.cwd(), "./public"),
+						dev: {
+							ip: "localhost",
+							port: 1234,
+							inspector_port: 5678,
+							local_protocol: "https",
+							upstream_protocol: "https",
+							host: "127.0.0.0",
+						},
+						send_metrics: true,
+
+						/** INHERITABLE ENVIRONMENT FIELDS **/
+						name: "pages-is-awesome-preview",
+						compatibility_date: "2024-01-01",
+						compatibility_flags: ["COMPATIBILITY_FLAG_preview"],
+						limits: { cpu_ms: 11 },
+						placement: { mode: "off" },
+
+						/** NON-INHERITABLE ENVIRONMENT FIELDS **/
+						vars: { VAR: "TEST_VAR-preview" },
+						durable_objects: {
+							bindings: [
+								{
+									name: "TEST_DO_BINDING",
+									class_name: "TEST_DO_CLASS_preview",
+									script_name: "TEST_DO_SCRIPT_NAME_preview",
+								},
+							],
+						},
+						kv_namespaces: [
+							{
+								binding: "TEST_KV_BINDING",
+								id: "TEST_KV_ID_preview",
+							},
+						],
+						queues: {
+							producers: [
+								{
+									binding: "TEST_QUEUE_PRODUCER_BINDING",
+									queue: "TEST_QUEUE_PRODUCER_NAME_preview",
+								},
+							],
+						},
+						r2_buckets: [
+							{
+								binding: "TEST_R2_BINDING",
+								bucket_name: "TEST_R2_BUCKET_NAME_preview",
+							},
+						],
+						d1_databases: [
+							{
+								binding: "TEST_D1_BINDING",
+								database_id: "TEST_D1_DB_ID_preview",
+								database_name: "TEST_D1_DB_NAME_preview",
+							},
+						],
+						vectorize: [
+							{
+								binding: "TEST_VECTORIZE_BINDING",
+								index_name: "TEST_VECTORIZE_INDEX_NAME_preview",
+							},
+						],
+						hyperdrive: [
+							{
+								binding: "TEST_HYPERDRIVE_BINDING",
+								id: "TEST_HYPERDRIVE_ID_preview",
+							},
+						],
+						services: [
+							{
+								binding: "TEST_SERVICE_BINDING",
+								service: "TEST_SERVICE_NAME_preview",
+							},
+						],
+						analytics_engine_datasets: [
+							{
+								binding: "TEST_AED_BINDING",
+							},
+						],
+						ai: {
+							binding: "TEST_AI_BINDING",
+						},
+					})
+				);
+
+				/*========================*/
+				/*      "production"      */
```

Comment:
Is there a reason these are in one test case, rather than a test case per environment?

Reply:
I had these separated at first, but as I was writing them, felt like a bit of overkill. Since these tests have basically the same setup, and test for the same thing, with the only diff being the env name, I figured this grouping would make more sense. But perhaps we lost some readability and tidiness in the process. I'll regroup

Reply:
✅